common/on-chip-calib.cpp

```diff
@@ -1139,6 +1128,18 @@ namespace rs2
         {
             try
             {
+                //Save options that are going to change during the calibration
+                if (_sub->s->supports( RS2_OPTION_EMITTER_ENABLED ))
+                {
+                    laser_status_prev = _sub->s->get_option( RS2_OPTION_EMITTER_ENABLED );
+                    _sub->s->set_option( RS2_OPTION_EMITTER_ENABLED, 0.0f );
+                }
+                if (_sub->s->supports( RS2_OPTION_THERMAL_COMPENSATION ))
+                {
+                    thermal_loop_prev = _sub->s->get_option( RS2_OPTION_THERMAL_COMPENSATION );
+                    _sub->s->set_option( RS2_OPTION_THERMAL_COMPENSATION, 0.f );
+                }
+
```

Comment:
By moving the code here those set/get options will be executed every iteration cycle (this is part of the main loop)- this may affect performance.
Imho, the best option is to create kind of LUT that for each state will define the state of laser/AE/Thermal loop controls, and that we'll use some 'enter_state(calib_1) /exist_state(calib_1)'

Reply:
This is called only once
Later on `calibrate` calls `run_on_chip_calibration` and inside that function implementation there is a loop